### PR TITLE
New Update (Salt, Token gen)

### PR DESCRIPTION
**Pull request by nyzd**

### CODE REVIEW

File: controllers/account/token.ts [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
-    for (let index = 0; index < input.length; index++) {
-        array.push(input.charCodeAt(index));
-    }
-
-    return Uint32Array.from(array);
-}
-
-class TokenGenerator {
-    readonly input: Uint8Array;
+class Token {
```
Comment (al-abd) [inline controllers/account/token.ts:5]: Rename to userId

File: controllers/account/token.ts [hunk 2]
```diff
@@ -1,33 +1,25 @@
...
+    constructor(input: number) {
         this.input = input;
         this.token = new Uint8Array();
     }
 
     public async generate(): Promise<this> {
+        const salt = new Salt(32)
+            .dateAsString()
+            .string(this.input.toString())
+            .randomBytes();
```
Comment (al-abd) [inline controllers/account/token.ts]: Put 32 here

File: lib/salt.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+class Salt {
+    private result: Uint32Array;
+    private usedSize: number;
```
Comment (al-abd) [inline lib/salt.ts]: What is this?

File: lib/salt.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        for (let i = 0; i < size; i++) {
+            const random = Math.floor(Math.random() * 0xFF);
+            this.pushToResult(random);
+        }
+
+        return this;
+    }
+
+    public number(number: number): this {
```
Comment (al-abd) [inline lib/salt.ts]: Delete this func

File: lib/salt.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        this.pushToResult(number);
+
+        return this;
+    }
+
+    public string(string: string): this {
```
Comment (al-abd) [inline lib/salt.ts]: Explain this

File: lib/test/controller_test.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { assertEquals } from "./mod.ts";
```
Comment (al-abd) [inline lib/test/controller_test.ts:1]: Std???

**Review by al-abd — APPROVED** [review]
Check comments

**Comment by nyzd** [comment]
I think if we substitute Salt for Bytes its be good. So we dont toch result of salt or create another Array.

**Comment by al-abd** [comment]
So what does token do except hashing?

And which food do we add salt to?

**Review by nyzd — COMMENTED** [review]
nema

**Pull request merged**